### ISSUE TITLE
训练参数内的seed与实际模型生成的seed不匹配

### ISSUE THREAD

**Issue by morettt** [modelscope/swift#1582]
我用来训练的配置指令，其中seed被我设置为了666：

#!/bin/bash
source activate swift

CUDA_VISIBLE_DEVICES=0 \
swift sft \
    --model_id_or_path /root/autodl-tmp/llama3.1-8B-chat \
    --dataset "你的数据集" \
    --num_train_epochs 6 \
    --model_type llama3_1-8b-instruct \
    --learning_rate 5e-4 \
    --gradient_accumulation_steps 16 \
    --lora_target_modules ALL \
    --sft_type lora \
    --save_steps 40 \
    --lora_alpha 32 \
    --lora_rank 8 \
    --seed 666 \
    --gradient_checkpointing true \
    --batch_size 1 \
    --weight_decay 0.1 \
    --add_output_dir_suffix False \
    --output_dir /root/swift/output/wsw

其中打印中也是显示seed=666

![image](https://github.com/user-attachments/assets/8c05bfa3-c7de-40dc-839c-8f236b17bc86)

但是真正被加载的时候却还是显示默认的seed=42

![image](https://github.com/user-attachments/assets/13bf773f-214f-41f4-96a4-4f95e40dcc7e)

**Comment by morettt** [comment]
拉取了项目，但是运行后并没有发现有改变,依旧是这个配置：

#!/bin/bash
source activate swift

CUDA_VISIBLE_DEVICES=0 \
swift sft \
    --model_id_or_path /root/autodl-tmp/Qwen2-7B-Chat \
    --dataset /root/swift/train.json \
    --num_train_epochs 6 \
    --model_type qwen2-7b-instruct \
    --learning_rate 5e-4 \
    --gradient_accumulation_steps 16 \
    --lora_target_modules ALL \
    --sft_type lora \
    --save_steps 40 \
    --lora_alpha 32 \
    --lora_rank 8 \
    --seed 666 \
    --gradient_checkpointing true \
    --batch_size 1 \
    --weight_decay 0.1 \
    --add_output_dir_suffix False \
    --output_dir /root/swift/output/23

还是显示42  

![image](https://github.com/user-attachments/assets/664e1b42-c4be-4399-b3b1-f8586c4b9a17)

![image](https://github.com/user-attachments/assets/f18d7459-7773-4708-aac1-995400e7f594)

**Comment by morettt** [comment]
> main分支已修复

拉取了项目，但是运行后并没有发现有改变,依旧是这个配置：

#!/bin/bash
source activate swift

CUDA_VISIBLE_DEVICES=0
swift sft
--model_id_or_path /root/autodl-tmp/Qwen2-7B-Chat
--dataset /root/swift/train.json
--num_train_epochs 6
--model_type qwen2-7b-instruct
--learning_rate 5e-4
--gradient_accumulation_steps 16
--lora_target_modules ALL
--sft_type lora
--save_steps 40
--lora_alpha 32
--lora_rank 8
--seed 666
--gradient_checkpointing true
--batch_size 1
--weight_decay 0.1
--add_output_dir_suffix False
--output_dir /root/swift/output/23

还是显示42

image

image
